feat: add accessor protocol support and refactor `stats/base/nanmin`

rahulptl165 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: passed
  - task: lint_package_json status: na
  - task: lint_repl_help status: passed
  - task: lint_javascript_src status: passed
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: passed
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: passed
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: passed
  - task: lint_typescript_tests status: passed
  - task: lint_license_headers status: passed ---

Resolves #5659 .

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds new `lib/accessors.js` file.
-   Modified `lib/nanmin.js` and `lib/ndarray.js` .
-   Modified bechmarks, tests, example and docs files accordingly.  

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #5659

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

rahulptl165 committed:
feat: add accessor protocol and refactor stats/base/nanmin

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

rahulptl165 commented:
Hi @kgryte, Please review my PR. thanks

rahulptl165 committed:
test: ensure to include existing assertions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/README.md:
```diff
@@ -36,33 +36,29 @@ limitations under the License.
-#### nanmin( N, x, stride )
+#### nanmin( N, x, strideX )
 
 Computes the minimum value of a strided array `x`, ignoring `NaN` values.
```
```suggestion
Computes the minimum value of a strided array, ignoring `NaN` values.
```

kgryte committed:
docs: update description

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/README.md:
```diff
@@ -72,42 +68,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
+-   **offsetX**: starting index for `x`.
 
-While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to calculate the minimum value for every other value in `x` starting from the second value
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameters support indexing semantics based on a starting indices. For example, to calculate the minimum value for every other value in `x` starting from the second value
```
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter support indexing semantics based on a starting index. For example, to calculate the minimum value for every other element in `x` starting from the second element
```

kgryte committed:
docs: fix grammar

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/README.md:
```diff
@@ -72,42 +68,35 @@ Note that indexing is relative to the first index. To introduce an offset, use [
+-   **offsetX**: starting index for `x`.
 
-While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying `buffer`, the `offset` parameter supports indexing semantics based on a starting index. For example, to calculate the minimum value for every other value in `x` starting from the second value
+While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter support indexing semantics based on a starting index. For example, to calculate the minimum value for every other element in `x` starting from the second element
```
```suggestion
While [`typed array`][mdn-typed-array] views mandate a view offset based on the underlying buffer, the offset parameter supports indexing semantics based on a starting index. For example, to calculate the minimum value for every other element in `x` starting from the second element
```

kgryte committed:
docs: fix grammar

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/README.md:
```diff
@@ -133,22 +123,12 @@ var v = nanmin.ndarray( N, x, 2, 1 );
-        x[ i ] = round( (randu()*100.0) - 50.0 );
-    }
-}
+var x = discreteUniform( 10, -50, 50, {
```
How you refactored this example does not preserve the original logic. See, e.g., https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/stats/strided/dnanmax#examples

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/docs/repl.txt:
```diff
@@ -1,9 +1,9 @@
 
-    The `N` and `stride` parameters determine which elements in `x` are accessed
-    at runtime.
+    The `N` and stride parameters determine which elements in the strided arrays
```
```suggestion
    The `N` and stride parameters determine which elements in the strided array
```

There is only one strided array.

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/examples/index.js:
```diff
@@ -18,22 +18,12 @@
-		x[ i ] = round( (randu()*100.0) - 50.0 );
-	}
-}
+var x = discreteUniform( 10, -50, 50, {
```
Same comment as README.

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/lib/accessors.js:
```diff
@@ -0,0 +1,95 @@
+/**
+* Computes the minimum value of a strided array, ignoring `NaN` values.
+*
+* @param {PositiveInteger} N - number of indexed elements
```
```suggestion
* @private
* @param {PositiveInteger} N - number of indexed elements
```

This function is not part of the package's public API.

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/lib/nanmin.js:
Would you mind renaming this file to `main.js` and updating `index.js` accordingly?

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/lib/nanmin.js:
```diff
@@ -41,46 +41,8 @@ var isNegativeZero = require( '@stdlib/math/base/assert/is-negative-zero' );
 * var v = nanmin( N, x, 1 );
```
Let's go ahead and remove L39 and replace `N` here with `x.length`. The use of `N` in this example doesn't really add anything.

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/lib/ndarray.js:
```diff
@@ -44,33 +46,38 @@ var isNegativeZero = require( '@stdlib/math/base/assert/is-negative-zero' );
 * var v = nanmin( N, x, 2, 1 );
```
You need to replace usage of `floor` in this example.

kgryte replied:
That may also apply to `index.js`.

kgryte commented on lib/node_modules/@stdlib/stats/base/nanmin/test/test.nanmin.js:
```diff
@@ -21,7 +21,7 @@
+var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var isNegativeZero = require( '@stdlib/math/base/assert/is-negative-zero' );
 var Float64Array = require( '@stdlib/array/float64' );
```
L28 will need to be updated after renaming `nanmin.js`. May apply elsewhere, as well.

kgryte committed:
docs: update note

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (commented):
Left an initial round of comments.

rahulptl165 committed:
refactor: addressed requested changes

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

rahulptl165 commented:
Hi @kgryte, I have addressed requested changes.

kgryte committed:
Merge branch 'develop' into feat/accessor-nanmin

Signed-off-by: Athan <kgryte@gmail.com>
